src/irmin-pack/inode.ml

```diff
@@ -327,24 +327,65 @@ struct
               t.tree <- Some x;
               x)
 
-    let rec list_entry ~find acc = function
-      | Empty -> acc
-      | Inode i -> list_values ~find acc (get_tree ~find i)
+    let rec drop n (l : 'a Seq.t) () =
+      match l () with
+      | l' when n = 0 -> l'
+      | Nil -> Nil
+      | Cons (_, l') -> drop (n - 1) l' ()
+
+    let rec take n (l : 'a Seq.t) () =
+      if n = 0 then Seq.Nil
+      else
+        match l () with Nil -> Nil | Cons (x, l') -> Cons (x, take (n - 1) l')
```

Comment:
Nit: slightly more efficient to have `take` return a `list` since we're forcing it anyway & we always convert to a list afterwards (AFAICS).

Reply:
sounds good -- can you add a commit doing this?